jme3-alloc/src/main/java/com/jme3/alloc/gc/GarbageCollectibleBuffers.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2009-2023 jMonkeyEngine
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.alloc.gc;
+
+import java.lang.ref.ReferenceQueue;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import com.jme3.alloc.util.NativeBufferUtils;
+
+/**
+ * A collection utility of GC-registered-direct buffers, the collection list is provided through
+ * the registered direct buffers addresses, direct buffers are added to a ReferenceQueue that is 
+ * registered to the GC as a part of post-mortem actions.
+ * 
+ * @author pavl_g
+ */
+public final class GarbageCollectibleBuffers {
+    private static final Logger logger = Logger.getLogger(GarbageCollectibleBuffers.class.getName());
+    private final Map<Long, GarbageCollectibleBuffer> bufferAddresses = new HashMap<>();
```

Comment:
As said on the forum, you should use a ConcurrentHashMap for `bufferAddresses`.

Reply:
But, why? we already synchronize the allocate/release on `ConcurrentNativeBufferAllocator`, so all calls here are synchronized by the  `ConcurrentNativeBufferAllocator`.

Reply:
Because MemoryScavenger thread calls `collectibles.deallocate()`.

Reply:
I wonder what other calls might be synchronized with this?

Reply:
If we are going to synchronize the `GarbageCollectibles` HashMap, then I think its better to synchronize the whole coding blocks by adding synchronized on the register and deallocate or at least the deallocate calls, but I need to find a strong reason to do this and understand what are the 2 parties that we are trying to synchronize here.

Reply:
@Ali-RS I synchronized the `deallocate` stacks using monitor locks instead, they are now thread-safe, I will mark this as resolved, if in doubt, you can always re-open the discussion.